core/modules/qdisp/JobQuery.cc

```diff
@@ -0,0 +1,124 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// System headers
+
+// Third-party headers
+
+// Class header
+#include "qdisp/JobQuery.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "qdisp/Executive.h"
+#include "qdisp/QueryRequest.h"
+#include "qdisp/QueryResource.h"
+
+namespace lsst {
+namespace qserv {
+namespace qdisp {
+
+void logErr(std::string const& msg, JobQuery* jq) {
+    std::ostringstream os;
+    os << msg << " " << *jq;
+    LOGF_ERROR("%1%" % os.str());
+}
+
+/** Attempt to run the job on a worker.
+ * @return - false if it can not setup the job or the maximum number of retries has been reached.
+ */
+bool JobQuery::runJob() {
+    LOGF_DEBUG("runJob %1%" % toString());
+    if (!_executive) {
+        logErr("runJob failed _executive=NULL", this);
+        return false;
+    }
+    bool cancelled = _executive->getCancelled();
+    bool handlerReset = _jobDescription.respHandler()->reset();
+    if (!cancelled && handlerReset) {
+        Ptr thisPtr = _weakThis.lock();
+        if (!thisPtr) {
```

Comment:
I do not understand how this could happen. Isn't it equivalent to `this == nullptr`?

If you need to make shared pointer from `this` then standard library has support for it already, `std::enable_shared_from_this<T>`